texastribune/tacobots
# Update security vulnerability alerts with Airtable

## What's this PR do?
This PR updates the Security Vulnerability Alerts Slackbot to implement the following changes: 
- It now creates a record in a test airtable base with information about the dependabot alert
- The local version and the version hosted up in the Cloud function are the same file/code
- Updates the requirements.txt file to include importing 'requests'
- The current date is added to the summary in the airtable record
- Removes the old version of the security vulnerabilities app
- Updates the README.md

## Why are we doing this? How does it help us?
- This Slackbot helps us manage requests by creating a record for them in Airtable.  

## How should this be manually tested?
This update uses Google Functions Framework in order to test locally. It requires three environment variables that should be placed in a .env file: SLACK_API_TOKEN, AIRTABLE_TOKEN, and AIRTABLE_BASE_ID.

The SLACK_API_TOKEN will be generated through the Slackbot API.
The AIRTABLE_TOKEN can be generated using the following instructions: https://support.airtable.com/docs/how-do-i-get-my-api-key
The AIRTABLE_BASE_ID can be found in the Airtable API.

The following command will create a server on localhost: functions-framework --target main --debug

After the server is running, there are two ways to test the Slackbot:

1. Curl Post Request
2. Running the invoke_function.py file

Further detail is provided in the README.md

## How should this change be communicated to end users?

## Are there any smells or added technical debt to note?

## What are the relevant tickets?

## Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


* [ ] Added migrations? *( )*

## TODOs / next steps:
* [ ] Add functions-framework command to Makefile
* [ ] Switch out the Airtable User ID token with a Personal API token
* [ ] Reference the Airtable record URL in the Slack notification
